### PR TITLE
-es-linting and code cleanup/refactoring for better maintainability

### PR DESCRIPTION
**Pull request by stopal-r7**
No functionality change aimed at this PR. Only linting and cleanup such as breaking logger module down to different modules, better usage of es6 classes etc.

### CODE REVIEW

**Comment by StephenHynes7** [comment]
Kinda hard to review ;) But looks ok.

**Comment by danilosterrapid7** [comment]
@stopal-r7 check travis, it seems that your code is not running cause the interpreter cannot identify es6 syntax.

**Comment by pquinn-r7** [comment]
:+1: thanks for changing the node versions tested

File: src/bunyanstream.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * BunyanStream class.
+ */
+class BunyanStream extends Writable {
+  constructor(opts) {
+    super({
+      objectMode: true
+    });
+
+    const loggerOpts = _.clone(opts || {});
```
Comment (danilosterrapid7) [inline src/bunyanstream.js]: I would rather use "Object.assign" than a third party library to clone an object. ;)

**Comment by danilosterrapid7** [comment]
👍 
Btw, the comment is not mandatory once you only restyle the code.

**Pull request merged**